### ISSUE TITLE
composer:Could not find package hiqdev/yii2-hiart-curl

### ISSUE THREAD

**Issue by imaxwen** [hiqdev/yii2-hiart#7]
Is this package alternatived by `yii2-hiart-guzzle` ?

**Comment by hiqsol** [comment]
Yes, you can choose which transport you prefer, see Configuration section
in README.

I think the problem is that yii2-hiart-curl has no releases, I will do one
today.
Please, show me your composer.json to be sure.

On Feb 7, 2017 9:23 AM, "Max.wen" <notifications@github.com> wrote:

Is this package alternatived by yii2-hiart-guzzle ?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/hiqdev/yii2-hiart/issues/7>, or mute the thread

**Comment by imaxwen** [comment]
@hiqsol  thanks for your reply, I just run `composer require yii2-hiart-curl` in command line, so the composer.json file not includes this package.

**Comment by hiqsol** [comment]
I've made release. Try again, should be ok now.

**Comment by imaxwen** [comment]
`yii2-hiart-curl` can be required by composer now, thanks.

if I just want to use `yii2-hiart-guzzle` as transport, I just need to require to `yii2-hiart-guzzle` into my project, right?

and Is there any more documents?

I've already done all configurations according to the `docs/*.md`.

currently I import the `yii2-hiart-guzzle` into my yii2 project, a model bellow:

```php
use hiqdev\hiart\ActiveRecord;

/**
 * denning file model
 * Class File
 * @package frontend\models
 */
class File extends ActiveRecord
{

    public function attributes()
    {
        return [
            'systemNo',
            'MRTA',
            'ODLoan',
        ];
    }

    public static function primaryKey()
    {
        return ['systemNo'];
    }


}
```

Usage in my controller:

```php

class TestController extends BaseController
{

    public function actionIndex()
    {
        $model = File::findOne(['systemNo' => '2000-0077']);

        var_dump($model);
    }
}

```


And I got an error: `Can not instantiate hiqdev\hiart\ConnectionInterface.`.

maybe I missed something, so if there is a full example will be very useful.  
thanks very much

**Comment by SilverFire** [comment]
> and Is there any more documents?

Unfortunately, no. The library is still under development. In addition to the documentation, available in docs directory, you can use PHPDocs for classes and methods. We will add more documentation in future.

> And I got an error: Can not instantiate hiqdev\hiart\ConnectionInterface..

We use [hiqdev/composer-config-plugin](https://github.com/hiqdev/composer-config-plugin) to merge configs automatically in our projects. In order to make this extension work without this plugin, you have to add its configuration to your application configuration. Full configuration you can find in [src/config/common.php](https://github.com/hiqdev/yii2-hiart/blob/master/src/config/common.php).

To fix the aforementioned error, you have to configure the [Dependency Injection Container](http://www.yiiframework.com/doc-2.0/guide-concept-di-container.html) in order to assosicate `hiqdev\hiart\ConnectionInterface` with a Connection class. See highlighted lines in [common.php](https://github.com/hiqdev/yii2-hiart/blob/master/src/config/common.php#L31-L37) and add them to your application config. Replace `Yii::$app->params['hiart.dbname']` with your `hiart` component name.

> if I just want to use yii2-hiart-guzzle as transport, I just need to require to yii2-hiart-guzzle into my project, right?

In case you don't use [hiqdev/composer-config-plugin](https://github.com/hiqdev/composer-config-plugin), you have to set the `requestClass` property of the `hiart` component, as it is advised in [docs/Configuration.md](https://github.com/hiqdev/yii2-hiart/blob/master/docs/readme/Configuration.md).

**Comment by hiqsol** [comment]
Hmm... Actually I wanted this simple configuration to work:

```php
    'components' => [
        'hiart' => [
            'class' => \hiqdev\hiart\rest\Connection::class,
            'requestClass' => \hiqdev\hiart\curl\Request::class,
            'baseUri' => 'https://site.com/api/v3/',
        ],
    ],
```

Without dependency injection configuration.

I will fix it. This simple config will work.

But still you should pay attention to `src/config/common.php` for it contains config for HiArt debug panel - which is very handy to debug API interactions.

**Comment by imaxwen** [comment]
@hiqsol thanks. 
Will the configuration in `application/config/main.php` overwrite those in `src/config/common.php`?

**Comment by hiqsol** [comment]
It depends on the order you merge it.

`src/config/common.php` is not intended for direct merging into your config, it's for composer-config-plugin which is another story and has no documentation now :(

So for simplicity just put everything to your main config.

I will make release soon that will allow simple config to work.

**Comment by imaxwen** [comment]
thanks ,can't wait for your next release 😂

**Comment by hiqsol** [comment]
Released 0.1.2
